vcd/datasource_vcd_catalog_vapp_template_test.go

```diff
@@ -0,0 +1,184 @@
+//go:build catalog || ALL || functional
+// +build catalog ALL functional
+
+package vcd
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+// Test catalog and vApp Template data sources.
+// Using a catalog data source we reference a vApp Template data source.
+// Using a vApp Template data source we create another vApp Template.
```

Comment:
Why creating a new resource? this is a data source test. You can use the default vApp template from NSX-V and NSX-T catalogs

Reply:
To be honest I followed same pattern as `catalog_item` data source tests, with a few additions. Probably this is not a good reason 😄 
Will change it.

Reply:
Indeed, it returns a task which I was ignoring. I have updated go-vcloud-director accordingly and the quirk is gone.